ogc/wmts-layer/src/main/java/com/esri/samples/wmts_layer/WmtsLayerSample.java

```diff
@@ -60,15 +60,15 @@ public void start(Stage stage) {
       // create a WMTS service from a URL
       String serviceURL = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer/WMTS";
       wmtsService = new WmtsService(serviceURL);
-      wmtsService.addDoneLoadingListener(() -> {
-        if (wmtsService.getLoadStatus() == LoadStatus.LOADED) {
+      wmtsService.loadStatusProperty().addListener((observable, oldValue, newValue) -> {
+        if (newValue == LoadStatus.LOADED) {
           WmtsServiceInfo wmtsServiceInfo = wmtsService.getServiceInfo();
           // get the first layer's ID
           List<WmtsLayerInfo> layerInfos = wmtsServiceInfo.getLayerInfos();
           // create the WMTS layer with the LayerInfo
           WmtsLayer wmtsLayer = new WmtsLayer(layerInfos.get(0));
           map.setBasemap(new Basemap(wmtsLayer));
-        } else {
+        } else if (newValue == LoadStatus.NOT_LOADED || newValue == LoadStatus.FAILED_TO_LOAD) {
```

Comment:
tempted to keep this as FAILED_TO_LOAD check. if the service is LOADING then we don't need an alert, but if the service is NOT LOADED that means it hasn't triggered a load which would be a code responsibility elsewhere. This way we can keep the alert notification clear with what the status of the load is. 

```suggestion
        } else if (newValue == LoadStatus.FAILED_TO_LOAD) {
```

Reply:
Fixed.

Reply:
Changed.